### ISSUE TITLE
Error building HDF5.jl package on Raspberry Pi

### ISSUE THREAD

**Issue by sairus7** [JuliaIO/HDF5.jl#588]
I've installed Julia language on Raspberry Pi as mentioned here:
https://www.raspberrypi.org/blog/julia-language-raspberry-pi/

But when trying to `]add HDF5` the following error appears:

```
(v1.2) pkg> add HDF5
 Resolving package versions...
 Installed CMakeWrapper ─────── v0.2.3
 Installed OrderedCollections ─ v1.1.0
 Installed Blosc ────────────── v0.5.1
 Installed BinDeps ──────────── v0.8.10
 Installed Parameters ───────── v0.12.0
 Installed BinaryProvider ───── v0.5.6
 Installed Compat ───────────── v2.2.0
 Installed HDF5 ─────────────── v0.12.4
 Installed CMake ────────────── v1.1.2
 Installed URIParser ────────── v0.4.0
  Updating ~/.julia/environments/v1.2/Project.toml
  [f67ccb44] + HDF5 v0.12.4
  Updating ~/.julia/environments/v1.2/Manifest.toml
  [9e28174c] + BinDeps v0.8.10
  [b99e7846] + BinaryProvider v0.5.6
  [a74b3585] + Blosc v0.5.1
  [631607c0] + CMake v1.1.2
  [d5fb7624] + CMakeWrapper v0.2.3
  [34da2185] + Compat v2.2.0
  [f67ccb44] + HDF5 v0.12.4
  [bac558e1] + OrderedCollections v1.1.0
  [d96e819e] + Parameters v0.12.0
  [30578b45] + URIParser v0.4.0
  [2a0f44e3] + Base64 
  [ade2ca70] + Dates 
  [8bb1440f] + DelimitedFiles 
  [8ba89e20] + Distributed 
  [b77e0a4c] + InteractiveUtils 
  [76f85450] + LibGit2 
  [8f399da3] + Libdl 
  [37e2e46d] + LinearAlgebra 
  [56ddb016] + Logging 
  [d6f4376e] + Markdown 
  [a63ad114] + Mmap 
  [44cfe95a] + Pkg 
  [de0858da] + Printf 
  [3fa0cd96] + REPL 
  [9a3f8284] + Random 
  [ea8e919c] + SHA 
  [9e88b42a] + Serialization 
  [1a1011a3] + SharedArrays 
  [6462fe0b] + Sockets 
  [2f01184e] + SparseArrays 
  [10745b16] + Statistics 
  [8dfed614] + Test 
  [cf7118a7] + UUIDs 
  [4ec0a83e] + Unicode 
  Building CMake → ~/.julia/packages/CMake/nSK2r/deps/build.log
  Building Blosc → ~/.julia/packages/Blosc/lzFr0/deps/build.log
  Building HDF5 ─→ ~/.julia/packages/HDF5/rF1Fe/deps/build.log
┌ Error: Error building HDF5: 
│ [ Info: Downloading https://github.com/bicycle1885/ZlibBuilder/releases/download/v1.0.4/Zlib.v1.2.11.arm-linux-gnueabihf.tar.gz to /home/pi/.julia/packages/HDF5/rF1Fe/deps/usr/downloads/Zlib.v1.2.11.arm-linux-gnueabihf.tar.gz...
│ ERROR: LoadError: LoadError: MethodError: no method matching iterate(::Nothing)
│ Closest candidates are:
│   iterate(!Matched::Core.SimpleVector) at essentials.jl:604
│   iterate(!Matched::Core.SimpleVector, !Matched::Any) at essentials.jl:604
│   iterate(!Matched::ExponentialBackOff) at error.jl:214
│   ...
│ Stacktrace:
│  [1] append_any(::Any, ::Vararg{Any,N} where N) at ./essentials.jl:722
│  [2] top-level scope at /home/pi/.julia/packages/HDF5/rF1Fe/deps/build_HDF5.v1.10.5.jl:40
│  [3] include at ./boot.jl:328 [inlined]
│  [4] include_relative(::Module, ::String) at ./loading.jl:1094
│  [5] include(::Module, ::String) at ./Base.jl:31
│  [6] include(::String) at ./client.jl:431
│  [7] top-level scope at /home/pi/.julia/packages/HDF5/rF1Fe/deps/build.jl:2
│  [8] include at ./boot.jl:328 [inlined]
│  [9] include_relative(::Module, ::String) at ./loading.jl:1094
│  [10] include(::Module, ::String) at ./Base.jl:31
│  [11] include(::String) at ./client.jl:431
│  [12] top-level scope at none:5
│ in expression starting at /home/pi/.julia/packages/HDF5/rF1Fe/deps/build_HDF5.v1.10.5.jl:40
│ in expression starting at /home/pi/.julia/packages/HDF5/rF1Fe/deps/build.jl:2
```

**Comment by sairus7** [comment]
Also see the discussion: https://discourse.julialang.org/t/julia-on-raspberry-pi-error-building-hdf5-package

**Comment by musm** [comment]
I think this is fixed you might need to pin HDF5_jll to v1.10.5

**Comment by musm** [comment]
Alternatively, you can use the system provided binaries (as long as they are new enough). See instructions:  https://github.com/JuliaIO/HDF5.jl#installation
